src/core/Rule.ml

```diff
@@ -66,17 +69,20 @@ end = struct
 
   let to_string x = x
 
-  (* TODO: re-enable validation after checking that we're not
-        introducing major breakage.
-        We may need to support other characters such as '@'
-        as used in Node.js file paths.
-        See https://github.com/returntocorp/semgrep/pull/8038
+  let validate =
+    let rex = SPcre.regexp "^[a-zA-Z0-9._-]*$" in
+    fun str -> SPcre.pmatch_noerr ~rex str
+
+  let sanitize_string str =
+    let buf = Buffer.create (String.length str) in
```

Comment:
you could use a regexp replace, like you did in Python

Reply:
sure, but are there benefits? I'd have to look up the function calls and make sure the regexp is compiled only once (which I didn't do it for Python). A decent `String` module would provide `String.filter` and `String.replace` in the standard library and I'd use one of these.